stock_picking_backorder_link/models/stock_picking.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2024 ForgeFlow S.L. (http://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def get_moves_to_cancel_from_backorders(self, move):
+        self.ensure_one()
+        moves = self.env["stock.move"]
+        if move.move_orig_ids:
+            moves_done = move.move_orig_ids.filtered(lambda m: m.state == "done")
+            done_previous_moves_qty = sum(m.product_qty for m in moves_done)
+            if done_previous_moves_qty >= move.product_qty:
+                moves = move.move_orig_ids.filtered(
+                    lambda m: m.state not in ["done", "cancel"]
+                )
+        return moves
+
+    def _action_done(self):
+        if self.env.company.stock_picking_backorder_link and self.env.context.get(
```

Comment:
IMHO, getting the param from picking (`company_id`) is more appropriate

Reply:
I've done some changes! Thx!